server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -289,44 +289,21 @@ public static class Builder extends FieldMapper.Builder {
 
         private final boolean withinMultiField;
 
-        public Builder(String name, IndexAnalyzers indexAnalyzers, boolean isSyntheticSourceEnabled) {
-            this(name, IndexVersion.current(), indexAnalyzers, isSyntheticSourceEnabled, false);
+        public Builder(String name, IndexAnalyzers indexAnalyzers) {
+            this(name, IndexVersion.current(), indexAnalyzers, false);
         }
 
-        public Builder(
-            String name,
-            IndexVersion indexCreatedVersion,
-            IndexAnalyzers indexAnalyzers,
-            boolean isSyntheticSourceEnabled,
-            boolean withinMultiField
-        ) {
+        public Builder(String name, IndexVersion indexCreatedVersion, IndexAnalyzers indexAnalyzers, boolean withinMultiField) {
             super(name);
 
-            // If synthetic source is used we need to either store this field
-            // to recreate the source or use keyword multi-fields for that.
-            // So if there are no suitable multi-fields we will default to
-            // storing the field without requiring users to explicitly set 'store'.
-            //
-            // If 'store' parameter was explicitly provided we'll reject the request.
-            // Note that if current builder is a multi field, then we don't need to store, given that responsibility lies with parent field
             this.withinMultiField = withinMultiField;
-            this.store = Parameter.storeParam(m -> ((TextFieldMapper) m).store, () -> {
```

Comment:
we no longer need to do this. `TextFieldMapper` now decides whether we need an extra `StoredField` during parsing.

Reply:
Yes, but I wonder if there are any customers that rely on `store = true` by default. We initially set it that way to support synthetic source. However, given thats done elsewhere now, we might be able to release this without too many concerns. I'll discuss with the team.